common-theme/layouts/partials/fetch-paginated-issues.html

```diff
@@ -0,0 +1,22 @@
+{{/* # This is a recursive partial
+  # The point is to walk the paginated issues
+  # So we don't lose coursework tasks pushed off page 1 as pull requests start piling up
+*/}}
+{{/* set the current page we are getting */}}
+{{ $issuesUrl := print .issuesUrlBase .page }}
+{{/* fetch the issues */}}
+{{ $pageOfIssues := getJSON $issuesUrl .headers }}
+{{/* get the issues we have so far in our Scratch */}}
+{{ $fetchedIssues := .Scratch.Get "issues" }}
+
+{{ if and $pageOfIssues (gt (len $pageOfIssues) 0) }}
+  {{ range $pageOfIssues }}
+    {{ $fetchedIssues = $fetchedIssues | append . }}
+  {{ end }}
+  {{ .Scratch.Set "issues" $fetchedIssues }}
+  {{/* If we got 100 issues, there might be more. Let's fetch the next page */}}
+  {{ if eq (len $pageOfIssues) 100 }}
```

Comment:
I'm not sure this is correct, I think GitHub will return non-100-item pages but still need pagination - we could maybe just always try to get page+1 and see if it has results?

The way my little proxy does it is to look in the `link` header for a `next` link, I'm not sure if there's a handy way to get at headers from a hugo response, but I would assume not?

Reply:
Hrm, it looks like there's a new way https://gohugo.io/functions/resources/getremote/#remote-data in the latest version and I can maybe get into them that way. Will look

Reply:
Coming back to say - it looks like I can't get full details yet https://github.com/gohugoio/hugo/issues/12521 

When this comes out I can do it that way and I will. In any case I will update to the new way of getting json in advance.

If a request is set to get 100 and there are fewer than 100 remaining, in what scenario will GitHub paginate? (Genuine question -- help me think through this so I can sort this for MCB)

Reply:
And yy re your point above: GH does treat PRs as issues, so that response contains both and we filter afterwards

Reply:
> If a request is set to get 100 and there are fewer than 100 remaining, in what scenario will GitHub paginate?

I'm not sure, but I'm pretty sure I've observed it. If you can convince yourself it doesn't ever do this, I won't argue :)

Reply:
I'm not convinced it won't happen, I just don't know _how_ it would happen. 

Also I think we might have to start versioning this platform... 🤮

Reply:
I'm going to have to push ahead with this because MCB are about to launch